src/OneinchCallee.sol

```diff
@@ -0,0 +1,163 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+pragma experimental ABIEncoderV2;
+
+interface GemJoinLike {
+    function dec() external view returns (uint256);
+
+    function gem() external view returns (TokenLike);
+
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external;
+
+    function transfer(address, uint256) external;
+
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface IAggregationExecutor {
+    function callBytes(address msgSender, bytes calldata data) external payable;
+}
+
+interface OneinchRouter {
+    struct SwapDescription {
+        TokenLike srcToken;
+        TokenLike dstToken;
+        address srcReceiver;
+        address dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+    }
+
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+contract OneinchCallee {
+    DaiJoinLike public daiJoin;
+    TokenLike public dai;
+    uint256 public constant RAY = 10**27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, 'ds-math-add-overflow');
+    }
+
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, 'ds-math-sub-underflow');
+    }
+
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _add(x, _sub(y, 1)) / y;
+    }
+
+    constructor(address daiJoinAddress) public {
+        daiJoin = DaiJoinLike(daiJoinAddress);
+        dai = daiJoin.dai();
+        dai.approve(daiJoinAddress, 2**256 - 1);
+    }
+
+    function _fromWad(address gemJoin, uint256 wad) internal view returns (uint256 amt) {
+        amt = wad / 10**(_sub(18, GemJoinLike(gemJoin).dec()));
+    }
+
+    function clipperCall(
+        address sender, // Clipper caller, pays back the loan
+        uint256 owe, // Dai amount to pay back          [rad]
+        uint256 slice, // Gem amount received           [wad]
+        bytes calldata data // Extra data, see below
+    ) external {
+        (
+            address to, // address to send remaining DAI to
+            address gemJoinAddress, // gemJoin adapter address
+            uint256 minProfit, // minimum profit in DAI to make [wad]
+            address router, // tx.to address received from the 1inch API
+            bytes memory oneinchData // reencoded 1inch parameters
+        ) = abi.decode(data, (address, address, uint256, address, bytes));
```

Comment:
I would keep the charter manager code here.  We may have one in the future and this will just work with it out-of-the-box if we add one.  Same with any keeper code you write.